module-apps/apps-common/ApplicationCommon.cpp

```diff
@@ -650,6 +652,32 @@ namespace app
         return sys::msgHandled();
     }
 
+    sys::MessagePointer ApplicationCommon::handleAppShutdownRefresh(sys::Message *msgl)
+    {
+        auto *msg = static_cast<AppShutdownRefreshMessage *>(msgl);
+        assert(msg);
+        if (windowsStack.isEmpty()) {
+            LOG_ERROR("Current window is not defined");
+            return sys::msgNotHandled();
+        }
+        if (windowsStack.isEmpty() || (getCurrentWindow()->getName() != msg->getWindowName())) {
+            LOG_DEBUG("Ignore shutdown refresh request for window %s we are on window %s",
+                      msg->getWindowName().c_str(),
+                      windowsStack.isEmpty() ? "none" : getCurrentWindow()->getName().c_str());
+            return sys::msgNotHandled();
+        }
```

Comment:
1. the First check is not that window is not defined. It's about if there is something else on the windows stack. I would say that adding a window builder to the application is closer to the window handling definition, than the created window on the stack.
2. This check is redundant, first, you check if isEmpty, then if is empty *or* (where you already checked isEmpty so it cant be empty)